Task4/js/main.js

```diff
@@ -0,0 +1,29 @@
+function disp(){
+	'use strict'
+
+	let inNum = prompt('数字を入力してください');
+	let outNum = 0;
+
+	//再帰処理
+	const loop = inNum => {
+		let lpNum = parseInt(inNum,10);
+		if (lpNum !== 0){
+			outNum += lpNum;
+		return loop(lpNum - 1);
+		} else {
+			return 1;
+		}
+	};
+
+	//メイン処理
+	if ( inNum === null || inNum === ''){
+		console.warn('入力が空かキャンセルされました');
+	} else if (inNum.match(/^([1-9]\d*|0)$/)){
+		loop(inNum);
```

Comment:
グローバルに定義した`outNum`を`loop`関数のなかから直接操作してますけど、再起処理でこういうことは避けた方が良いです。

以下の2点の縛りで結果が同じになるように、`loop`関数の書き方を工夫してみてください。
* 5行目の`outNum`の宣言は削除
* `loop`関数の呼び出し時に`outNum`を`const`で宣言

```suggestion
		const outNum = loop(inNum);
```